Update OpenSearch Java Client documentation: update version, add ApacheHttpClient5Transport, fix code samples (#3510)
opensearch-project/documentation-website: _clients/java.md

This getting started guide illustrates how to connect to OpenSearch, index documents, and run queries. For the client source code, see the [opensearch-java repo](https://github.com/opensearch-project/opensearch-java).

## Installing the client using Apache HttpClient 5 Transport
To start using the OpenSearch Java client, you need to provide the transport. The default one is `ApacheHttpClient5TransportBuilder` and comes with the Java client.

```xml
<dependency>
  <groupId>org.opensearch.client</groupId>
  <artifactId>opensearch-java</artifactId>
  <version>2.3.0</version>
</dependency>
```
{% include copy.html %}

> If you're using Gradle, add the following dependencies to your project.

Reviewer: ```suggestion
If you're using Gradle, add the following dependencies to your project:
```

```
dependencies {
  implementation 'org.opensearch.client:opensearch-java:2.3.0'
}
```
{% include copy.html %}

You can now start your OpenSearch cluster.

## Installing the client using RestClient Transport

> The alternative way to create a Java client is by using `RestClient` based transport, ensure that you have the following dependencies in your project's `pom.xml` file:

Reviewer: ```suggestion
Alternatively, you can create a Java client by using the `RestClient`-based transport. In this case, make sure that you have the following dependencies in your project's `pom.xml` file:
```
